### PR TITLE
Merge pull request #93 from Azure/azuresovereigncloud

### PR DESCRIPTION
**Pull request by arjgupta**
GitHub Actions Beta Preview for deploying to Azure Stack Hub

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -9,6 +9,10 @@ inputs:
     description: 'Set this value to true to enable Azure PowerShell Login in addition to Az CLI login'
     required: false
     default: false
+  environment:
+    description: 'Name of the environment. Supported values are azurecloud, azurestack, azureusgovernment, azurechinacloud, azuregermancloud. Default being azurecloud'
```
Comment (asranja) [inline action.yml:13]: casing in description doesnt match the defaultValue's casing.

File: src/PowerShell/Utilities/ScriptBuilder.ts [hunk 2]
```diff
@@ -9,6 +9,9 @@ export default class ScriptBuilder {
         let command = `Clear-AzContext -Scope Process;
              Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue;`;
         if (scheme === Constants.ServicePrincipal) {
+            if (args.environment.toLowerCase() == "azurestack") {
+                command += `Add-AzEnvironment -Name ${args.environment} -ARMEndpoint ${args.resourceManagerEndpointUrl} | out-null;`;
+            }
```
Comment (asranja) [inline src/PowerShell/Utilities/ScriptBuilder.ts:14]: nit: leave a line

File: src/main.ts [hunk 3]
```diff
@@ -38,15 +45,58 @@ async function main() {
...
         if (!subscriptionId && !allowNoSubscriptionsLogin) {
             throw new Error("Not all values are present in the creds object. Ensure subscriptionId is supplied.");
         }
+        
+       if (!azureSupportedCloudName.has(environment)){
+            throw new Error("Unsupported value for environment is passed.The list of supported values for environment are ‘azureusgovernment', ‘azurechinacloud’, ‘azuregermancloud’, ‘azurecloud’ or ’azurestack’");
+       }
+    
+        // Attempting Az cli login
+        if (environment == "azurestack") {
```
Comment (asranja) [inline src/main.ts:66]: we could have a utility for azure stack

**Review by asranja — APPROVED** [review]

File: src/main.ts (hunk 3 above)
Comment (arjgupta) [reply src/main.ts:66]: I cannot do anything for this now. these changes are already in master. I'll add a todo and leave it for later.

File: action.yml (hunk 1 above)
Comment (arjgupta) [reply action.yml:13]: done.

File: src/PowerShell/Utilities/ScriptBuilder.ts (hunk 2 above)
Comment (arjgupta) [reply src/PowerShell/Utilities/ScriptBuilder.ts:14]: done.

**Pull request merged**